src/Compilers/Core/Portable/Operations/BasicBlock.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    /// <summary>
+    /// PROTOTYPE(dataflow): Add documentation
+    /// </summary>
+    public enum BasicBlockKind
+    {
+        Entry,
+        Exit,
+        Block
+    }
+
+    /// <summary>
+    /// PROTOTYPE(dataflow): Add documentation
+    /// </summary>
+    public sealed class BasicBlock
+    {
+        private ImmutableArray<IOperation>.Builder _statements;
+        private ImmutableHashSet<BasicBlock>.Builder _predecessors;
+
+        public BasicBlock(BasicBlockKind kind)
+        {
+            Kind = kind;
+            _statements = ImmutableArray.CreateBuilder<IOperation>();
+            _predecessors = ImmutableHashSet.CreateBuilder<BasicBlock>();
+        }
+
+        public BasicBlockKind Kind { get; private set; }
+        public ImmutableArray<IOperation> Statements => _statements.ToImmutable();
+        public (IOperation Condition, bool JumpIfTrue, BasicBlock Destination) Conditional { get; internal set; }
```

Comment:
This seems the first time we use a tuple in public API. Should this be a dedicated/lightweight type instead?

Reply:
Agreed. I don't like the idea of having a Tuple in the public API.

---
In reply to: [162138392](https://github.com/dotnet/roslyn/pull/24263#discussion_r162138392) [](ancestors = 162138392)

Reply:
Added a comment

---
In reply to: [162194804](https://github.com/dotnet/roslyn/pull/24263#discussion_r162194804) [](ancestors = 162194804,162138392)